postcommit.py

```diff
@@ -0,0 +1,642 @@
+
+    def getRemovedFiles(self, branch):
+
+        return self.repo.status(self.branch_details[branch]['parent'], branch, match=None)[2]
+
+    def getAllFiles(self, changeset):
+
+        return self.repo[changeset].manifest().keys()
+
+    def getChangeContext(self, changeset):
+
+        return self.repo[changeset]
+
+#	***************************************************************************************************************
+#	getDiffOpts :
+#
+#			Provide extra options for diff command
+#
+#       *************************************************************************************************************** 
+    def getDiffOpts(self):
+        options = {'unified': '0', 'ignore_space_change': True, 'ignore_all_space': True}
+        diffoptions = patch.diffopts(self.ui, options)
+        return diffoptions
+
+#	***************************************************************************************************************
+#	getDiffContent :
+#
+#			Show changes between two changeset
+#
+#			@param : changeset1,changeset2
+#			
+#			@return : [{'changed_lines': [1,4,6], 'file': 'test'}]
+#       *************************************************************************************************************** 
+    def getDiffContent(self, changeset1, changeset2):
+        diffcontent = []
+        diffcontent = patch.diff(self.repo, changeset1, changeset2, opts=self.getDiffOpts())
+        diffcontent = list(diffcontent)
+        diff_changes = self.parseDiffContent(diffcontent)
+
+        return diff_changes
+
+
+#	***************************************************************************************************************
+#	getDiffLines :
+#
+#			Find number of lines newly added in file
+#
+#			@param : changes
+#
+#			@return : [1,2,4,5]
+#       ***************************************************************************************************************
+    def getDiffLines(self,changes):
+        difflines = []
+        for val in changes:
+            changed_line_values = str(val)
+
+            if changed_line_values.find(',') > -1:
+                start_linenumber = int(changed_line_values.split(',')[0])
+                total_changed_lines = int(changed_line_values.split(',')[1])
+                for line in range(start_linenumber, start_linenumber + total_changed_lines):
+                    difflines.append(line)
+            else:
+                difflines.append(int(changed_line_values))
+
+        return difflines
+
+#       ***************************************************************************************************************
+#	parseDiffContent :
+#
+#			Parse diff contents for two changesets
+#
+#			@param : diff
+#			
+#			@return : [{'changed_lines': [1,4,6], 'file': 'test'}]
+#       *************************************************************************************************************** 
+    def parseDiffContent(self, diff):
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename method "parseDiffContent" to match the regular expression ^[a-z_][a-z0-9_]{2,}$. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS100)